launch/launch/frontend/parser.py

```diff
@@ -86,28 +97,68 @@ def parse_description(self, entity: Entity) -> LaunchDescription:
         actions = [self.parse_action(child) for child in entity.children]
         return LaunchDescription(actions)
 
+    @classmethod
+    def get_available_extensions(cls) -> List[Text]:
+        """Return the registered extensions."""
+        cls.load_parser_implementations()
+        return cls.frontend_parsers.keys()
+
+    @classmethod
+    def is_extension_valid(
+        cls,
+        extension: Text,
+    ) -> bool:
+        """Return an entity loaded with a markup file."""
+        cls.load_parser_implementations()
+        return extension in cls.frontend_parsers
+
+    @classmethod
+    def get_parser_from_extension(
+        cls,
+        extension: Text,
+    ) -> Optional[Type['Parser']]:
+        """Return an entity loaded with a markup file."""
+        cls.load_parser_implementations()
+        try:
+            return cls.frontend_parsers[extension]
+        except KeyError:
+            raise RuntimeError('Not recognized frontend implementation')
+
     @classmethod
     def load(
         cls,
         file: Union[str, io.TextIOBase],
     ) -> (Entity, 'Parser'):
-        """Return an entity loaded with a markup file."""
+        """
+        Parse an Entity from a markup language-based launch file.
+
+        Parsers are exposed and provided by available frontend implementations.
+        To choose the right parser, it'll first attempt to infer the launch
+        description format based on the filename extension, if any.
+        If format inference fails, it'll try all available parsers one after the other.
+        """
+        # Imported here, to avoid recursive import.
         cls.load_parser_implementations()
+
+        def get_key(extension):
+            def key(x):
+                return x[0] != extension
+            return key
+        exceptions = []
+        extension = ''
         if is_a(file, str):
```

Comment:
@ivanpauno fyi: you may also get the extension if its a `file` object read from the filesystem.